pkg/internal/tracking/context.go

```diff
@@ -0,0 +1,57 @@
+package tracking
+
+import (
+	"context"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+)
+
+const (
+	ProviderVersion string = "v0.98.0" // TODO(SNOW-1814934): Currently hardcoded, make it computed
+	MetadataPrefix  string = "terraform_provider_usage_tracking"
+)
+
+type key int
+
+var metadataContextKey key
+
+type Operation string
```

Comment:
I think we need to agree how the enums should look like, in this case I would leave it as is because that is how the majority of our enums look like. Also, like with resources.Resource I would need to hold it inside the struct as plain string.